webapp/components/wpt-amend-metadata.js

```diff
@@ -356,6 +365,20 @@ class AmendMetadata extends LoadingState(PathInfo(ProductInfo(PolymerElement)))
     }
   }
 
+  checkEnableTriageButton() {
```

Comment:
The setup (used with `on-input`) with this method is rather inefficient. This function is called for every letter typed in any of the input field above. And each time, this function goes through `displayedMetadata`. Ideally this function should only called once for each input field.

Reply:
I agree. I was having trouble finding an approach that can check if all fields are filled for a dynamic and unbounded number of input fields that have a state tracked by object pr. Reading about the [event objects model property](https://polymer-library.polymer-project.org/2.0/docs/devguide/templates) gave a better idea for a more succinct solution.

Reply:
Per offline discussions: 

In terms of an efficient way to check it, I have two approaches in mind - you can use a use counter + onchange (https://www.w3schools.com/jsref/event_onchange.asp), and data bind the use counter to a function that checks if the use counter === the number of input fields.

The second one should be easier - instead of disabling triage button, we can always enable the triage button and check that all input fields are filled when clicks it. If they are not, we prompt user to fill them without closing the dialog. In this case, we don't have to check all input fields dynamically